src/components/Lotion.vue

```diff
@@ -95,6 +95,17 @@ function setBlockType (blockIdx: number, type: BlockType) {
 function merge (blockIdx: number) {
   if (blockIdx === 0) return
 
+  // When deleting the first character of non-text block
+  // the block should first turn into a text block
+  if([BlockType.H1, BlockType.H2, BlockType.H3,BlockType.Quote]
+      .includes(props.page.blocks[blockIdx].type)){
+    setBlockType(blockIdx, BlockType.Text)
+    setTimeout(()=>{
+      blockElements.value[blockIdx].moveToStart()
+    })
+    return
+  }
+
```

Comment:
This looks good but can we move it to above line 96? Otherwise it wouldn't apply to the first block.

Reply:
That's right!

Reply:
LGTM!